numpy/random/mtrand/mtrand.pyx

```diff
@@ -1799,16 +1799,16 @@ cdef class RandomState:
 
         fshape = PyFloat_AsDouble(shape)
         if not PyErr_Occurred():
-            if fshape <= 0:
-                raise ValueError("shape <= 0")
+            if fshape < 0:
+                raise ValueError("shape < 0")
             return cont1_array_sc(self.internal_state, rk_standard_gamma,
                                   size, fshape, self.lock)
 
         PyErr_Clear()
         oshape = <ndarray> PyArray_FROM_OTF(shape, NPY_DOUBLE,
                                             NPY_ARRAY_ALIGNED)
-        if np.any(np.less_equal(oshape, 0.0)):
-            raise ValueError("shape <= 0")
+        if np.any(np.less(oshape, 0.0)):
+            raise ValueError("shape < 0")
         return cont1_array(self.internal_state, rk_standard_gamma, size,
```

Comment:
This is calling `rk_standard_gamma` with `shape = 0`. If you go to [the source code of that function](https://github.com/numpy/numpy/blob/master/numpy/random/mtrand/distributions.c#L125) in `distributions.c`, you will find that, with `shape = 0`, you are computing things like `X = pow(U, 1./shape))`. I think we should handle that particular case directly there, not rely on the behavior of `pow` for corner cases.